src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -80,13 +81,17 @@ public class TeamExplorerContext : ITeamExplorerContext
             gitExt.ActiveRepositoriesChanged += Refresh;
         }
 
-        void StartRefresh() => JoinableTaskFactory.Context.Factory.RunAsync(QueueRefreshAsync).Task.FileAndForget(log);
-        void Refresh() => JoinableTaskFactory.Context.Factory.Run(QueueRefreshAsync);
+        void StartRefresh() => JoinableTaskFactory.RunAsync(QueueRefreshAsync).Task.FileAndForget(log);
+        void Refresh() => JoinableTaskFactory.Run(QueueRefreshAsync);
 
         async Task QueueRefreshAsync()
         {
-            await JoinableTaskCollection.JoinTillEmptyAsync();
-            await JoinableTaskFactory.RunAsync(RefreshAsync);
+            if (refreshJoinableTask != null)
+            {
+                await refreshJoinableTask.JoinAsync(); // make sure StartRefresh has completed
```

Comment:
You can drop the `.JoinAsync()` call. It's implied any time you await a `JoinableTask`.